hibernate-core/src/main/java/org/hibernate/id/enhanced/SequenceStyleGenerator.java

```diff
@@ -234,6 +237,37 @@ public void configure(Type type, Properties params, ServiceRegistry serviceRegis
 		final int initialValue = determineInitialValue( params );
 		int incrementSize = determineIncrementSize( params );
 
+		if ( isPhysicalSequence( jdbcEnvironment, forceTableUse ) ) {
+			String databaseSequenceName = sequenceName.getObjectName().getText();
+			Long databaseIncrementValue = getSequenceIncrementValue( jdbcEnvironment, databaseSequenceName );
+
+			if ( databaseIncrementValue != null && !databaseIncrementValue.equals( (long) incrementSize ) ) {
+				int dbIncrementValue = databaseIncrementValue.intValue();
+
+				SequenceMismatchStrategy sequenceMismatchStrategy = configurationService.getSetting(
+						AvailableSettings.SEQUENCE_INCREMENT_SIZE_MISMATCH_STRATEGY,
+						SequenceMismatchStrategy::interpret,
+						SequenceMismatchStrategy.EXCEPTION
+				);
+
+				switch ( sequenceMismatchStrategy ) {
+					case EXCEPTION:
+						throw new MappingException(
+								String.format(
+										"The increment size of the [%s] sequence is set to [%d] in the entity mapping " +
+												"while the associated database sequence increment size is [%d].",
+										databaseSequenceName, incrementSize, dbIncrementValue
+								)
+						);
+					case FIX:
+						incrementSize = dbIncrementValue;
+					case LOG:
+						LOG.sequenceIncrementSizeMismatch( databaseSequenceName, incrementSize, dbIncrementValue );
```

Comment:
dbIncrementValue is not used after this if code block and the log which says "The database sequence increment size will take precedence to avoid identifier allocation conflicts" seems incorrect